File: Python/searching/linear_search.py
def linear_search_iterative(arr: list, num: int) -> int:
    for i in range(0, len(arr)):
        if arr[i] == num:
            return i
    return -1


def __linear_search_recursive(arr: list, num: int, i: int) -> int:
    if i == len(arr):
        return -1
    if arr[i] == num:
        return i
    return __linear_search_recursive(arr, num, i + 1)


def linear_search_recursive(arr: list, num: int) -> int:
    return __linear_search_recursive(arr, num, 0)

File: Java/searching/LinearSearch.java
package searching;
public class LinearSearch {
    public static int linearSearchIterative(int[] arr, int num) {
        for (int i = 0; i < arr.length; i++)
            if (arr[i] == num)
                return i;
        return -1;
    }

    public static int linearSearchRecursive(int[] arr, int num) {
        return linearSearchRecursive(arr, num, 0);
    }

    private static int linearSearchRecursive(int[] arr, int num, int i) {
        if (i == arr.length)
            return -1;
        if (arr[i] == num)
            return i;
        return linearSearchRecursive(arr, num, i + 1);
    }
}
